src/frontend/classic/components/Viewer.svelte

```diff
@@ -40,11 +44,81 @@
     function toggleFullScreen() {
         if (!document.fullscreenElement) {
             document.documentElement.requestFullscreen();
-        } else {
-            if (document.exitFullscreen) {
-                document.exitFullscreen();
+        } else if (document.exitFullscreen) {
+            document.exitFullscreen();
+        }
+    }
+
+    function loadThumbnail(el: Element) {
+        const div = el as HTMLDivElement;
+        const src = div.getAttribute("data-image");
+
+        if (src) {
+            div.style.backgroundImage = `url('${src}')`;
+        }
+    }
+
+    function loadWide(el: Element) {
+        const img = el as HTMLImageElement;
+        const src = img.getAttribute("data-image");
+
+        if (src) {
+            img.src = src;
+        }
+    }
+
+    function loadImages(currentIndex: number): void {
+        const image = imagesToLoad[currentIndex];
+
+        setTimeout(() => {
+            if (image && image.getAttribute("data-mode") === "thumbnail") {
+                loadThumbnail(image);
+            } else if (image) {
+                loadWide(image);
+            }
+
+            if (currentIndex + 1 < imagesToLoad.length) {
+                loadImages(currentIndex + 1);
+            } else {
+                imagesToLoad = [];
             }
+        }, throttlingDelay);
+    }
+
+    function isImagesToLoadFull(): boolean {
+        return imagesToLoad.length === item.Entries.length;
+    }
+
+    function addElement(el: Element | null) {
+        if (el && el.getAttribute("data-index")) {
+            const index = Number(el.getAttribute("data-index"));
+            imagesToLoad[index] = el;
         }
+
+        if (isImagesToLoadFull()) {
+            loadImages(0);
+        }
+    }
+
+    function isThumbnailImageUtil(el: HTMLElement): boolean {
```

Comment:
there is no need to create such small function if there is no plan to reuse it.

Reply:
Same here, the clean code book recommend using this kind of small function but I'm ok with removing it

Reply:
Yes and No, for example, i had to scroll up to find the test done and go back. That was harder to read ;)
Alternative is to name the test within a variable
`const isThumbnailImageUtil: Boolean = myval-1 > 0;`

Reply:
Ok I see your point ^^, thanks for the teaching :)